app/src/main/java/com/evelynhasama/bucketbuds/CreateBucketFragment.java

```diff
@@ -0,0 +1,267 @@
+    }
+
+    public SaveCallback getStandardSaveCallback(String errorDescription){
+        return new SaveCallback() {
+            @Override
+            public void done(ParseException e) {
+                if (e != null) {
+                    Log.d(TAG, errorDescription + e);
+                }
+            }
+        };
+    }
+
+    public void addSaveBucketList(UserPub userPub){
+        userPub.addBucket(bucketList);
+        userPub.saveInBackground(getStandardSaveCallback("addSaveBucketList "));
+    }
+
+    // Trigger gallery selection for a photo
+    public void onPickPhoto(View view) {
+        Log.d(TAG, "onPickPhoto");
+        if (ActivityCompat.checkSelfPermission(getContext(), Manifest.permission.READ_EXTERNAL_STORAGE) != PackageManager.PERMISSION_GRANTED) {
+            ActivityCompat.requestPermissions(getActivity(), new String[]{Manifest.permission.READ_EXTERNAL_STORAGE, Manifest.permission.WRITE_EXTERNAL_STORAGE}, PICK_PHOTO_CODE);
+        } else {
+            Log.d(TAG, "startng Intent");
+            Intent galleryIntent = new Intent(Intent.ACTION_PICK, android.provider.MediaStore.Images.Media.EXTERNAL_CONTENT_URI);
+            startActivityForResult(galleryIntent, PICK_PHOTO_CODE);
+        }
+    }
+
+    public Bitmap loadFromUri(Uri photoUri) {
+        Bitmap image = null;
+        try {
+            // check version of Android on device
+            if(Build.VERSION.SDK_INT > 27){
+                // on newer versions of Android, use the new decodeBitmap method
+                ImageDecoder.Source source = ImageDecoder.createSource(getContext().getContentResolver(), photoUri);
+                image = ImageDecoder.decodeBitmap(source);
+            } else {
+                // support older versions of Android by using getBitmap
+                image = MediaStore.Images.Media.getBitmap(getContext().getContentResolver(), photoUri);
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return image;
+    }
+
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        if ((data != null) && requestCode == PICK_PHOTO_CODE) {
+            Uri photoUri = data.getData();
+
+            // Load the image located at photoUri into selectedImage
+            Bitmap selectedImage = loadFromUri(photoUri);
+            ivBucketImage.setImageBitmap(selectedImage);
+
+            // convert to ParseFile and set to BucketList
+            ByteArrayOutputStream stream = new ByteArrayOutputStream();
+            selectedImage.compress(Bitmap.CompressFormat.PNG,100,stream);
+            byte[] byteArray = stream.toByteArray();
+
+            ParseFile file = new ParseFile(PHOTO_FILE_NAME, byteArray);
+            bucketList.setImage(file);
+        }
+    }
+
+    @Override
+    public void onRequestPermissionsResult(int requestCode, @NonNull String permissions[], @NonNull int[] grantResults)
+    {
+        switch (requestCode) {
```

Comment:
nit: why a `switch` if there's only one condition?

Reply:
yeah that was unnecessary😔